api/v1beta1/azuremachine_default.go

```diff
@@ -174,11 +174,11 @@ func (s *AzureMachineSpec) SetNetworkInterfacesDefaults() {
 	}
 }
 
-// GetSubscriptionID returns the subscription ID for the given cluster name and namespace.
-func GetSubscriptionID(cli client.Client, clusterName string, namespace string, maxAttempts int) (string, error) {
+// GetOwnerCluster returns the owner cluster for the given cluster name and namespace.
+func GetOwnerCluster(cli client.Client, clusterName string, namespace string, maxAttempts int) (*clusterv1.Cluster, error) {
```

Comment:
Would it be better to return `ownerCluster.Namespace` and `ownerCluster.Spec.InfrastructureRef.Name` instead of a pointer to the cluster object ? One more advantage(I can think of) of returning string(in this case) over a pointer would be that the variable `ownerCluster := &clusterv1.Cluster{}` would not "escape" `GetOwnerCluster()` func, adding to the function optimization.

Reply:
That's a great suggestion! I initially left it as returning the whoel cluster in case this is useful elsewhere, but returning just the name and namespace makes more sense in this case. Will work on implementing this now!

Reply:
Also, I realized I should be returning `ownerCluster.Spec.InfrastructureRef.Namespace` instead because I'm looking for the `AzureCluster` and not the `Cluster`. They should be the same but I don't think it's guaranteed.